packages/@aws-cdk/aws-appsync/test/appsync-schema.test.ts

```diff
@@ -272,4 +283,108 @@ describe('testing schema definition mode `file`', () => {
       api.addSubscription('blah', new appsync.ResolvableField({ returnType: t.string }));
     }).toThrowError('Unable to add subscription. Schema definition mode must be CODE. Received: FILE');
   });
+});
+
+describe('testing schema definition mode `s3`', () => {
+
+  test('definition mode `s3` produces correct output', () => {
+    // GIVEN
+    const schemaAsset = new assets.Asset(stack, 'Schema', {
+      path: join(__dirname, 'appsync.test.graphql'),
+    });
+    // WHEN
+    new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location(schemaAsset.s3ObjectUrl),
```

Comment:
This is a weird test.. we have a separate `fromAsset()` static factory method for `Schema`, so why also use Asset here...?